libqtile/widget/currentlayout.py

```diff
@@ -239,6 +223,47 @@ def _setup_images(self):
 
         self.icons_loaded = True
 
-    def finalize(self):
-        self._remove_hooks()
-        base._TextBox.finalize(self)
+
+class CurrentLayoutIconOrText(CurrentLayoutIcon):
+    """
+    Either display the icon of the current layout or its name, imagine switching
+    between CurrentLayout and CurrentLayoutIcon with a click.
+    """
+
+    defaults = [
+        ("draw_icon_first", True, "Should draw icon or text when bar is initialized."),
```

Comment:
Is it possible to add this config to `CurrentLayout` and collapse the existing `CurrentLayoutIcon` into it as well, instead of adding a whole new widget?

Reply:
Beat me to it. We definitely don't want/need 3 layout widgets. 

I've got similar things in qtile-extras where you can choose icon or text modes or have text appear when you click on the icon (and then disappear after a timeout).

That approach would be good here but we may need a migration too if we remove a widget.